loki: switching to a streaming JSON parser

gabor opened the pull request:
WIP

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=45653) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=45653) before we can accept your contribution.<br/>**48** out of **49** committers have signed the CLA.<br/><br/>:white_check_mark: hugohaggmark<br/>:white_check_mark: leventebalogh<br/>:white_check_mark: kaydelaney<br/>:white_check_mark: mtanda<br/>:white_check_mark: tolzhabayev<br/>:white_check_mark: ashharrison90<br/>:white_check_mark: dsotirakis<br/>:white_check_mark: gelicia<br/>:white_check_mark: undef1nd<br/>:white_check_mark: baez90<br/>:white_check_mark: juschm<br/>:white_check_mark: FZambia<br/>:white_check_mark: wbrowne<br/>:white_check_mark: connorlindsey<br/>:white_check_mark: zuchka<br/>:white_check_mark: nvtkaszpir<br/>:white_check_mark: Elfo404<br/>:white_check_mark: andresmgot<br/>:white_check_mark: Jguer<br/>:white_check_mark: joanlopez<br/>:white_check_mark: zserge<br/>:white_check_mark: papagian<br/>:white_check_mark: jackw<br/>:white_check_mark: torkelo<br/>:white_check_mark: ying-jeanne<br/>:white_check_mark: toddtreece<br/>:white_check_mark: aocenas<br/>:white_check_mark: yuri-tceretian<br/>:white_check_mark: konrad147<br/>:white_check_mark: dprokop<br/>:white_check_mark: nathanrodman<br/>:white_check_mark: kminehart<br/>:white_check_mark: ivanahuckova<br/>:white_check_mark: ifrost<br/>:white_check_mark: tharun208<br/>:white_check_mark: perosb<br/>:white_check_mark: sdboyer<br/>:white_check_mark: gabor<br/>:white_check_mark: sunker<br/>:white_check_mark: ArturWierzbicki<br/>:white_check_mark: yaelleC<br/>:white_check_mark: yangkb09<br/>:white_check_mark: grobinson-grafana<br/>:white_check_mark: srclosson<br/>:white_check_mark: kylebrandt<br/>:white_check_mark: mellieA<br/>:white_check_mark: grafanabot<br/>:white_check_mark: kazeborja<br/>:x: renovate[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=45653) it.</sub>

leventebalogh commented:
@gabor Seems like Levitate was only complaining because the return type of `getAvailableIcons()` has changed, it shouldn't be breaking, please ignore it.

gabor committed:
converter: remove __name__ customization because Loki does not do that

Loki does not handle __name__ in a special way.
for Prometheus, the caller can implement the formatting
by themselves

gabor committed:
loki: backend: use streaming-json parser

gabor committed:
converter: change labels-formatting

the labels.String() method does not handle strange values well

gabor commented:
closing this PR, will create a new one.
